### PR TITLE
Add tests and docs for while loops

### LINKED ISSUE #749: Document loops and add some E2E tests
`while` loops work today but they need documentation in the Sway book and some E2E tests.

### PR DESCRIPTION
**Pull request by nfurfaro**
All in the title.
- adds a new e2e test in the sway repo.
- adds a new `Control Flow` section in the sway book.

closes #749

### CODE REVIEW

File: docs/src/control-flow/loops.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Loops in Sway are currently limited to `while` loops. This is what they look like:
+
+```sway
+while counter < 10 {
+    counter = counter + 1;
+}
+```
+
+You need the `while` keyword, some condition (`value < 10` in this case) which will be evaluated each cycle, and a block of code inside the curly braces (`{...}`) to execute each cycle.
```
Comment (mohammadfawaz) [inline docs/src/control-flow/loops.md:11]:
```suggestion
You need the `while` keyword, some condition (`value < 10` in this case) which will be evaluated each iteration, and a block of code inside the curly braces (`{...}`) to execute each iteration.
```

**Comment by nfurfaro** [comment]
hmm. This was passing locally. Will get it working here before review is needed.

File: test/src/e2e_vm_tests/test_programs/while_loops/src/main.sw [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    let mut counter3 = 0;
+    while counter2 < 10 {
+        if counter2 == 3 {
+            // ensure that condition is now invalid:
+            counter2 = 10;
+        } else {
+            counter2 = counter2 + 1;
+            counter3 = counter3 + 1;
+        }
+    }
```
Comment (mohammadfawaz) [inline test/src/e2e_vm_tests/test_programs/while_loops/src/main.sw:24]: Can we actually add a small example with nested while loops?

Comment (nfurfaro) [reply test/src/e2e_vm_tests/test_programs/while_loops/src/main.sw:24]: For sure. I'll add a test and an example in the docs.

**Review by mohammadfawaz — APPROVED** [review]
Awesome, thanks!

**Pull request merged**